src/core/text-buffer.cc

```diff
@@ -378,6 +383,138 @@ struct TextBuffer::Layer {
     return result;
   }
 
+  struct SubsequenceMatchVariant {
+    size_t query_index = 0;
+    std::vector<uint32_t> match_indices;
+    int16_t score = 0;
+  };
+
+  vector<SubsequenceMatch> find_words_with_subsequence(u16string query, const u16string &extra_word_characters) {
+    size_t query_index = 0;
+    Point position;
+    Point current_word_start;
+    u16string current_word;
+
+    std::transform(query.begin(), query.end(), query.begin(), std::towlower);
+
+    // First, find the start position of all words matching the given
+    // subsequence.
+    std::unordered_map<u16string, vector<Point>> substring_matches;
+
+    for_each_chunk_in_range(Point(), extent(), [&] (TextSlice chunk) -> bool {
+      for (uint16_t c : chunk) {
+        bool is_word_character =
+          std::iswalnum(c) ||
```

Comment:
Should we ` std::setlocale(LC_ALL, "en_US.utf8");` to support unicode?

See the example at http://en.cppreference.com/w/cpp/string/wide/iswalnum

Reply:
Yeah, if you want to submit a followup PR that enables unicode that works. Just a heads up though that I think the encoding is actually UTF-16. Each character is explicitly encoded by 2 bytes. We use this encoding because it's how v8 represents all strings internally.

Reply:
Good point about utf16. I'm also wondering what other side effects `setlocale` has. Need to check on that.